vcr/credential/validator.go

```diff
@@ -123,3 +123,87 @@ func (d nutsOrganizationCredentialValidator) Validate(credential vc.VerifiableCr
 
 	return nil
 }
+
+// nutsAuthorizationCredentialValidator checks for mandatory fields: id, legalBase, purposeOfUse.
+// It checks if the value for legalBase.consentType is either 'explicit' or 'implied'.
+// When 'explicit', both the evidence and subject subfields must be filled.
+// When 'implied', the restrictions array must have values.
+type nutsAuthorizationCredentialValidator struct{}
+
+func (d nutsAuthorizationCredentialValidator) Validate(credential vc.VerifiableCredential) error {
+	var target = make([]NutsAuthorizationCredentialSubject, 0)
+
+	if err := validate(credential); err != nil {
+		return err
+	}
+
+	if !credential.IsType(*NutsAuthorizationCredentialTypeURI) {
+		return failure(fmt.Sprintf("type '%s' is required", NutsAuthorizationCredentialType))
+	}
+
+	// if it fails, length check will trigger
+	_ = credential.UnmarshalCredentialSubject(&target)
+	if len(target) != 1 {
+		return failure("single CredentialSubject expected")
+	}
+	cs := target[0]
+
+	if cs.ID == "" {
+		return failure("'credentialSubject.ID' is nil")
+	}
+	if cs.PurposeOfUse == "" {
+		return failure("'credentialSubject.PurposeOfUse' is nil")
+	}
+	switch cs.LegalBase.ConsentType {
+	case "implied":
+		if len(cs.Restrictions) == 0 {
+			return failure("'credentialSubject.Restrictions' must have entries when consentType is 'implied'")
+		}
+	case "explicit":
+		if cs.Subject == nil || len(strings.TrimSpace(*cs.Subject)) == 0 {
+			return failure("'credentialSubject.Subject' is required when consentType is 'explicit'")
+		}
+		if cs.LegalBase.Evidence == nil {
+			return failure("'credentialSubject.LegalBase.Evidence' is required when consentType is 'explicit'")
+		}
+		if len(strings.TrimSpace(cs.LegalBase.Evidence.Path)) == 0 {
+			return failure("'credentialSubject.LegalBase.Evidence.Path' is required when consentType is 'explicit'")
+		}
+		if len(strings.TrimSpace(cs.LegalBase.Evidence.Type)) == 0 {
+			return failure("'credentialSubject.LegalBase.Evidence.Type' is required when consentType is 'explicit'")
+		}
+	default:
+		return failure("'credentialSubject.LegalBase.ConsentType' must be 'implied' or 'explicit'")
+	}
+
+	return validateRestrictions(cs.Restrictions)
+}
+
+var validOperationTypes = []string{"read", "vread", "update", "patch", "delete", "history", "create", "search"}
```

Comment:
I generally like to wrap mutable var that are actually consts (often slices) in a function so they can't accidentally be mutated (because a new instance is returned every time)